### PR TITLE
Compatitable with GMT0018 512 key and signature

### PR DESCRIPTION
**Pull request by MaggieNgWu**
use 512 key and signature

support no password

### CODE REVIEW

File: hsm/sdf/SDFCryptoProvider.cpp [hunk 1]
```diff
@@ -115,25 +108,37 @@ unsigned int SDFCryptoProvider::Sign(Key const& key, AlgorithmType algorithm,
...
+                SDF_ReleasePrivateKeyAccessRight(sessionHandle, key.identifier());
+            }
         }
-        else
+        else     
         {
             ECCrefPrivateKey eccKey;
-            eccKey.bits = 32 * 8;
-            memcpy(eccKey.D, key.privateKey()->data(), 32);
+            eccKey.bits = 256;
```
Comment (bxq2011hust) [inline hsm/sdf/SDFCryptoProvider.cpp:135]: use const variable instead of 256?

File: hsm/sdf/SDFCryptoProvider.cpp [hunk 2]
```diff
@@ -289,9 +295,9 @@ unsigned int SDFCryptoProvider::Verify(Key const& key, AlgorithmType algorithm,
         else
         {
             ECCrefPublicKey eccKey;
-            eccKey.bits = 32 * 8;
-            memcpy(eccKey.x, key.publicKey()->data(), 32);
-            memcpy(eccKey.y, key.publicKey()->data() + 32, 32);
+            eccKey.bits = 256;
+            memcpy(eccKey.x+32, key.publicKey()->data(), 32);
```
Comment (bxq2011hust) [inline hsm/sdf/SDFCryptoProvider.cpp:299]: make sure the 0-31 of eccKey.x/y is all set to 0

**Review by bxq2011hust — APPROVED** [review]

File: hsm/sdf/SDFCryptoProvider.cpp (hunk 1 above)
Comment (MaggieNgWu) [reply hsm/sdf/SDFCryptoProvider.cpp]: OK

File: hsm/sdf/SDFCryptoProvider.cpp (hunk 2 above)
Comment (MaggieNgWu) [reply hsm/sdf/SDFCryptoProvider.cpp:300]: are all 0

**Pull request merged**